### PR TITLE
Cw 488 seed offset

### PR DESCRIPTION
**Pull request by konstantinullrich**
# Description

- Add switcher to select between a BIP39 and Electrum Wallet.
- Add Passphrase option on wallet creation.
- Implements Electrum Passphrases based on the official implementation [here](https://github.com/spesmilo/electrum/blob/bd845080b852ab19c54c2c4bd5ec8a36255b9e35/electrum/mnemonic.py#L175)
- Hide Length-selector on advanced settings from restore flow

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

**Comment by konstantinullrich** [comment]
Still some visual kinks to workout. eg. Passphrase is missing when reentering the advanced settings. The Passphrase does not propagate to previous screen

File: lib/store/settings_store.dart [hunk 1]
```diff
@@ -587,6 +597,9 @@ abstract class SettingsStoreBase with Store {
   @observable
   SeedType moneroSeedType;
 
+  @observable
+  DerivationTypeSetting bitcoinDerivationType;
```
Comment (OmarHatem28) [inline lib/store/settings_store.dart:601]: **Minor:**
would be easier to rename it to bitcoinSeedType (everywhere)

**Review by OmarHatem28 — CHANGES_REQUESTED** [review]
let's change the UI as mentioned in ux channel
Seed type at the top, followed by the same UI components (exchange and fiat API settings)
then passphrase
then toggles

File: lib/view_model/advanced_privacy_settings_view_model.dart [hunk 2]
```diff
@@ -52,7 +52,14 @@ abstract class AdvancedPrivacySettingsViewModelBase with Store {
     }
   }
 
-  bool get hasSeedTypeOption => type == WalletType.monero || type == WalletType.wownero;
+  bool get hasSeedTypeOption => [WalletType.monero, WalletType.wownero].contains(type);
+
+  bool get hasPassphraseOption => [
```
[the remark this answers is not in the document]
Comment (konstantinullrich) [reply lib/view_model/advanced_privacy_settings_view_model.dart:57]: I added it to the credentials, but I did not implement it yet

File: lib/store/settings_store.dart (hunk 1 above)
Comment (konstantinullrich) [reply lib/store/settings_store.dart:601]: But it it does not change the seed type, but only the key derivation

File: lib/view_model/wallet_creation_vm.dart [hunk 3]
```diff
@@ -115,11 +117,23 @@ abstract class WalletCreationVMBase with Store {
...
+              .getElectrumDerivations()[DerivationType.bip39]!
+              .firstWhere((element) => element.description == "Standard BIP84 native segwit");
+        }
+        return bitcoin!.getElectrumDerivations()[DerivationType.electrum]!.first;
       case WalletType.litecoin:
+        if (useBip39) {
+          return bitcoin!
+              .getElectrumDerivations()[DerivationType.bip39]!
+              .firstWhere((element) => element.description == "Default Litecoin");
+        }
```
Comment (OmarHatem28) [inline lib/view_model/wallet_creation_vm.dart:136]: just hard code the derivation path here better than comparing with a description string

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**